src/Infrastructure/Province/DbalProvinceRepository.php

```diff
@@ -27,6 +28,13 @@ public function findAll(): PromiseInterface
     {
         return $this
             ->connection
-            ->findBy(self::TABLE);
+            ->findBy(self::TABLE)
+            ->then(function ($provinces) {
+                $provincesHidratation = [];
+                foreach ($provinces AS $province) {
+                    $provincesHidratation[] = new ProvinceProxy($province);
+                }
+                return \React\Promise\resolve($provincesHidratation);
```

Comment:
Returning a plan value inside a Promise callback means returning a promise as well. You can return here the array and should work properly as well :)

Reply:
This means you can write `return $provincesHidratation;` instead?